### ISSUE TITLE
deploy_node fails due to a warning in E2E tests

### ISSUE THREAD

**Issue by jbertran** [scality/metalk8s#1331]
**Component**: tests

**What happened**:

In CI (example build at https://eve.devsca.com/github/scality/metalk8s/#/builders/1/builds/6944):

```
k8s_client = <kubernetes.client.apis.core_v1_api.CoreV1Api object at 0x7f3de587b6d8>
command = ['salt-run', 'state.orchestrate', 'metalk8s.orchestrate.deploy_node', 'saltenv=metalk8s-2.1.0-dev', 'pillar={"orchestrate": {"node_name": "metalk8s-7hfyk-node-1.novalocal"}}']
ssh_config = '/home/centos/ssh_config'
    def run_salt_command(k8s_client, command, ssh_config):
        """Run a command inside the salt-master container."""
        name = 'salt-master-{}'.format(
            utils.resolve_hostname('bootstrap', ssh_config)
        )
        stderr = k8s.stream.stream(
            k8s_client.connect_get_namespaced_pod_exec,
            name=name, namespace='kube-system', container='salt-master',
            command=command,
            stderr=True, stdin=False, stdout=False, tty=False
        )
>       assert not stderr, 'deploy failed with {}'.format(stderr)
E       AssertionError: deploy failed with [WARNING ] WARNING: Ignoring DaemonSet-managed pods: kube-proxy-h7zgv
E         
E       assert not '[WARNING ] WARNING: Ignoring DaemonSet-managed pods: kube-proxy-h7zgv\n'
```

Best guess: occasionally this test runs when the `kube-proxy` daemonset is up, and reports this to the exec API call in `stderr`.

From a cursory search, there is no way to get a clean return code from an exec with the API.

**What was expected**: This test should not fail because of a warning.

**Steps to reproduce**

**Resolution proposal** (optional):

Either find a way to get a clean return code from the API, or check the contents of `stderr` for error lines with whatever heuristics seem to apply.

**Comment by gdemonet** [comment]
One could try to pass `-l error` to Salt

**Comment by gdemonet** [comment]
(was not automatically closed, the syntax used in commit msg was `Closes: #1305 #1331`, while it should have been `Closes: #1305 Closes: #1331`)
